services/scabbard/libscabbard/src/service/shared.rs

```diff
@@ -95,15 +106,84 @@ impl ScabbardShared {
         &self.coordinator_service_id
     }
 
-    pub fn add_batch_to_queue(&mut self, batch: BatchPair) {
+    /// Returns the number of pending batches
+    pub fn pending_batches_count(&self) -> usize {
+        self.batch_queue.len()
+    }
+
+    /// set whether we are accepting new batches
+    pub fn set_accepting_batches(&mut self, accepting: bool) {
+        self.accepting_batches = accepting;
+    }
+
+    /// Check whether the pending batch queue has either gotten too big and back pressure has
+    /// should be enabled.
+    /// If back pressure has been enabled, check to see if the queue has reduced in size enough
+    /// to start accepting batches again.
+    fn update_accepting_batches(&mut self) -> Result<(), ScabbardError> {
```

Comment:
There are two decisions coded within this function, but those decision points occur in different points of the code, and are unrelated in terms of the algorithm. This should be refactored.

For example, this function is called when adding a batch -- but then this function checks for a change in both increasing and decreasing batch counts. When adding a batch, it is unnecessary/incorrect to check for the decreasing case.

Reply:
I'll update so this code is directly in the add/pop batch queue functions instead

Reply:
updated, removes update_accepting_batches